skbio/metadata/_interval.py

```diff
@@ -530,38 +562,28 @@ def num_interval_features(self):
         '''The total number of interval features.'''
         return len(self._intervals)
 
-    def _rebuild_tree(method):
-        """Rebuild the IntervalTree."""
-        @functools.wraps(method)
-        def inner(self, *args, **kwargs):
-            if self._is_stale_tree is False:
-                return method(self, *args, **kwargs)
-            self._interval_tree = IntervalTree()
-            for f in self._intervals:
-                for start, end in f.bounds:
-                    self._interval_tree.add(start, end, f)
-            self._is_stale_tree = False
-            return method(self, *args, **kwargs)
-        return inner
-
     def _reverse(self):
         """Reverse ``IntervalMetadata`` object.
 
         This operation reverses all of the interval coordinates.
         For instance, this can be used to compare coordinates
         in the forward strand to coordinates in the reversal strand.
         """
+
         for f in self._intervals:
-            intvls = [(self.upper_bound - x[1], self.upper_bound - x[0])
-                      for x in reversed(f.bounds)]
+            try:
+                intvls = [(self.upper_bound - x[1], self.upper_bound - x[0])
+                          for x in reversed(f.bounds)]
+            except TypeError:
+                raise TypeError('You cannot reverse the coordinates '
+                                'when the upper bound is `None`')
             f.bounds = intvls
 
-        # DONT' forget this!!!
+        # DON'T forget this!!!
         self._is_stale_tree = True
 
     @classonlymethod
-    @experimental(as_of="0.5.1")
-    def concat(cls, interval_metadata):
+    def _concat(cls, interval_metadata):
```

Comment:
> because @ebolyen and I think the only use case of _concat is inside the sequence objects.

I thought you were arguing earlier against using private APIs in other parts of the codebase? Following that logic, the mixin would prefer to use `concat` over a private `_concat`.

> I don't see use case for merge() with upper bound of None yet. hesitate to add it. let's add it when necessary?

You updated `concat` to explicitly handle `upper_bound=None`. We need to make sure `merge` handles it appropriately too (they are similar operations), even if you don't have a use-case for it. Other users can try to `merge` with `upper_bound=None` and it needs to work correctly.